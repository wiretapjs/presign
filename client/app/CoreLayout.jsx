import { childrenPropTypes } from 'lib/util/commonPropTypes'
import React from 'react'
import View from 'app/View'
import { Footer } from 'lib/components'

const CoreLayout = (props) => (
  <View>
    {props.children}
    <Footer />
  </View>
)

CoreLayout.propTypes = {
  children: childrenPropTypes,
}

export default CoreLayout
