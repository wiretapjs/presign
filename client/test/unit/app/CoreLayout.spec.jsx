import React from 'react'
import { shallow } from 'enzyme'
import CoreLayout from 'app/CoreLayout.jsx'

describe('CoreLayout', () => {
  it('renders correctly', () => {
    const comp = shallow(<CoreLayout />)
    expect(comp).toMatchSnapshot()
  })
})
