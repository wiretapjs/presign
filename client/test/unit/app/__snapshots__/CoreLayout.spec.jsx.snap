// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CoreLayout renders correctly 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <View>
    <Footer />
  </View>,
  "nodes": Array [
    <View>
      <Footer />
    </View>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 2,
      "_context": Object {},
      "_currentElement": <CoreLayout />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": StatelessComponent {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": null,
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <View>
          <Footer />
        </View>,
        "_debugID": 2,
        "_renderedOutput": <View>
          <Footer />
        </View>,
      },
      "_renderedNodeType": 1,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <CoreLayout />,
}
`;
